src/helpers/tests/stackByAttribute.spec.ts

```diff
@@ -0,0 +1,166 @@
+                            measureHeaderItem: {
+                                name: "Amount",
+                                format: "$#,##0.00",
+                                localIdentifier: "82d26962672c436bb183a02a78551918",
+                                uri: "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/1279",
+                                identifier: "ah1EuQxwaCqs",
+                            },
+                        },
+                    ],
+                },
+            },
+        ],
+    },
+];
+const attributeHeaderItems: Execution.IResultHeaderItem[][][] = [
+    [
+        [
+            {
+                attributeHeaderItem: {
+                    name: "East Coast",
+                    uri: "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/1023/elements?id=1225",
+                },
+            },
+            {
+                attributeHeaderItem: {
+                    name: "West Coast",
+                    uri: "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/1023/elements?id=1237",
+                },
+            },
+        ],
+    ],
+    [],
+];
+
+const mdObject: VisualizationObject.IVisualizationObjectContent = {
+    buckets: [
+        {
+            localIdentifier: "measures",
+            items: [
+                {
+                    measure: {
+                        localIdentifier: "82d26962672c436bb183a02a78551918",
+                        definition: {
+                            measureDefinition: {
+                                item: { uri: "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/1279" },
+                            },
+                        },
+                        title: "Amount",
+                    },
+                },
+            ],
+        },
+        {
+            localIdentifier: "stack",
+            items: [
+                {
+                    visualizationAttribute: {
+                        localIdentifier: "02eed1089e6149d9b448637cf32b2c43",
+                        displayForm: { uri: "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/1024" },
+                    },
+                },
+            ],
+        },
+    ],
+    filters: [
+        {
+            positiveAttributeFilter: {
+                displayForm: { uri: "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/952" },
+                in: [
+                    "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/949/elements?id=168279",
+                    "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/949/elements?id=168282",
+                ],
+            },
+        },
+        {
+            negativeAttributeFilter: {
+                displayForm: { uri: "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/1024" },
+                notIn: [],
+            },
+        },
+    ],
+    visualizationClass: { uri: "/gdc/md/dfnkvzqa683mz1c29ijdkydrsodm8wjw/obj/75536" },
+};
+
+describe("getStackByAttribute", () => {
+    it.each([[VisualizationTypes.AREA], [VisualizationTypes.TREEMAP]])(
```

Comment:
Because `getStackByAttribute` has 2 cases:
`
if (isTreemap(type)) {
        ...
        return stackByAttribute;
}
...
return otherStackByAttribute;
`
I have `TREEMAP` for if case, and `AREA` for else case.
So we don't need `PIE`

Reply:
> I would not use `it.each` here. You are using the visualization type inside of the test to determine the values of multiple variables. Just extract the expected stackByAttribute into the constant outside of the test and write two different test cases for Area and TreeMap. They will be short and more readable.

Updated :)